src/net_processing.cpp

```diff
@@ -3206,6 +3206,10 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             return;
         }
 
+        // Also stop processing the transaction early if we are still in IBD since we don't
+        // have enough information to validate it yet. Not a protocol violation.
+        if (::ChainstateActive().IsInitialBlockDownload()) return;
```

Comment:
Any reason to add a silent merge conflict with dongcarl's work? Could simply use `m_chainman.ActiveChainstate().IsInitialBlockDownload()`

Reply:
Using the member also mirrors the check used in `NetMsgType::INV` processing.

Reply:
Could log the rejection to the debug NET category?

Reply:
> Could log the rejection to the debug NET category?

we considered adding logging, but it seemed like it could be annoying if you're connected to a client who keeps sending TXs? seems like if you get 1 you might get many?

Reply:
I think you will already log that a message of type `tx` has been received, so adding another line wouldn't make things worse than they are now?

My thinking is that this code path should rarely be hit, because outbound connections generally don't announce txs via raw `tx` messages and inbound connections would only connect to you once your address is well propagated on the network, in which case you'll have left IBD.

If a peer wants to annoy you, they can already send a message that will log at least one line in the NET debug category (before and after this pull).

Reply:
I also thought we'd go by this [comment](https://github.com/bitcoin/bitcoin/blob/master/src/logging.h#L164-L166) that says not to log too much unnecessarily. My goal here is to do as little as possible when there's peers sending us unsolicited transactions we have no use for (which apparently does happen).

Reply:
Ah, that comment is about the unconditional log, which probably shouldn't be used in net at all. The conditional NET log doesn't have that concern. (If it did, we already have a massive problem, because on a medium activity node the NET log category eats up about 1TB of disk every 6 months)

Reply:
@MarcoFalke seems like you think this information useful to surface to the user, can you share why? what could a node operator do with this information?

Reply:
As a transaction received during IBD is necessary unsolicited, you might want to `setban` the faultive peer because that's a hint of a peer non-compliant with the inv-getdata sequence ?

Reply:
You could use the "received: tx (x bytes) peer=x" [log](https://github.com/bitcoin/bitcoin/blob/b9f41df1ead4b6a83a51fc41966b111c8459c313/src/net_processing.cpp#L2527) if you wanted to look for tx messages in IBD

Reply:
Fair enough. The existing debug log statements are sufficient to determine whether this code path was hit. Though, for an easy `grep` I'd still prefer a specific log statement, but I guess I can add this myself.